src/main/java/com/streamr/client/dataunion/DataUnion.java

```diff
@@ -0,0 +1,233 @@
+        if(amount.equals(BigInteger.ZERO)) {
+            return new EthereumTransactionReceipt(sidechain.withdrawAll(new Address(member), new Bool(true)).send());
+        }
+        else {
+            return new EthereumTransactionReceipt(sidechain.withdraw(new Address(member), new Uint256(amount), new Bool(true)).send());
+        }
+    }
+
+    /**
+     *
+     * @param amount amout in wei or 0 to withdraw everything
+     * @return
+     * @throws Exception
+     */
+    public EthereumTransactionReceipt withdrawSelf(BigInteger amount) throws Exception {
+        return withdrawSelf(sidechainCred.getAddress(), amount);
+    }
+
+    /**
+     * withdraw your own balance to another address
+     *
+     * @param to
+     * @param amount amout in wei or 0 to withdraw everything
+     * @return
+     * @throws Exception
+     */
+    public EthereumTransactionReceipt withdrawSelf(String to, BigInteger amount) throws Exception {
+        if(amount.equals(BigInteger.ZERO)) {
+            return new EthereumTransactionReceipt(sidechain.withdrawAllTo(new Address(to), new Bool(true)).send());
+        }
+        else {
+            return new EthereumTransactionReceipt(sidechain.withdrawTo(new Address(to), new Uint256(amount), new Bool(true)).send());
+        }
+    }
+
+    /**
+     *
+     * @param withdrawerPrivateKey
+     * @param to
+     * @param amount amout in wei or 0 to withdraw everything
+     * @return
+     * @throws Exception
+     */
+    public EthereumTransactionReceipt withdraw(BigInteger withdrawerPrivateKey, String to, BigInteger amount) throws Exception {
+        return withdraw(Credentials.create(ECKeyPair.create(withdrawerPrivateKey)), to, amount);
+    }
+
+    /**
+     *
+     * @param withdrawerPrivateKey
+     * @param to
+     * @param amount amout in wei or 0 to withdraw everything
+     * @return
+     * @throws Exception
+     */
+    public EthereumTransactionReceipt withdraw(String withdrawerPrivateKey, String to, BigInteger amount) throws Exception {
+        return withdraw(Credentials.create(withdrawerPrivateKey), to, amount);
+    }
+
+    protected EthereumTransactionReceipt withdraw(Credentials from, String to, BigInteger amount) throws Exception {
+        byte[] req = createWithdrawRequest(from.getAddress(), to, amount);
+        byte[] sig = toBytes65(Sign.signPrefixedMessage(req, from.getEcKeyPair()));
+
+        if(amount.equals(BigInteger.ZERO)) {
+            return new EthereumTransactionReceipt(sidechain.withdrawAllToSigned(new Address(from.getAddress()), new Address(to), new Bool(true), new DynamicBytes(sig)).send());
+        }
+        else {
+            return new EthereumTransactionReceipt(sidechain.withdrawToSigned(new Address(from.getAddress()), new Address(to), new Uint256(amount), new Bool(true), new DynamicBytes(sig)).send());
+        }
+    }
+
+    public BigInteger totalEarnings() throws Exception {
```

Comment:
Could prefix all getters with "get".

Reply:
I was thinking about the same, but then I thought that Java convention states that getters should not produce any side effects. On lines 202-210 there are getters which produce side effects (which is surprising for a Java programmer). I think those methods should be renamed without 'get'.

Reply:
E.g. both `lifetimeMemberEarnings()` and `getWithdrawableEarnings()` fetch some amount of earnings, can't see why there should be a different naming.

Reply:
the get/no-get name difference is actually in Solidity. If I write a view method, I add "get" to the method name. But Soldity automatically generates getters that are the same as variable name. So if you make `uint public x`, Solidity generates a view function call `x()`

You want client getters to have different names that Solidity contract? I prefer that names match

Reply:
Yes, I prefer Java naming style. This library is a Java library so the default naming convention should be Java's. I guess the users of the library don't know anything about the Solidity contract. They are mostly just Java-developers who are used to have `get`-prefix for methods that fetch some info and don't change anything. 

If the name is not prefixed with `get`, it implies that there is some other things happening. My first understanding of these methods was that there is some Builder-pattern as it looks like that. Less surprises is better.